openforcefield/typing/engines/smirnoff/parameters.py

```diff
@@ -2489,12 +2551,117 @@ def check_handler_compatibility(self,
                                          tolerance_attrs=float_attrs_to_compare+unit_attrs_to_compare,
                                          tolerance=self._SCALETOL)
 
+
+    def assign_charge_from_molecules(self, molecule, charge_mols):
+        """
+        Given an input molecule, checks against a list of molecules for an isomorphic match. If found, assigns
+        partial charges from the match to the input molecule.
```

Comment:
This looks good, but wonder if in the future, it'd be nice to do LibraryCharges on fragments.

Reply:
This function is for handling the `charge_from_molecules` kwarg. It is not related to LibraryCharges (except that it gets higher priority in the charge method hierarchy). This code existed before the PR, it's just been moved into `ElectrostaticsHandler` because its priority was messed up when it was attached to `ToolkitAM1BCCHandler`